src/R/Components/Test/PackageManager/PackageManagerIntegrationTest.cs

```diff
@@ -0,0 +1,208 @@
+
+            // Validate some of the base packages
+            result.Should().NotBeEmpty();
+
+            var graphics = result.SingleOrDefault(pkg => pkg.Package == "graphics");
+            graphics.Should().NotBeNull();
+            graphics.LibPath.Should().NotBeNullOrWhiteSpace();
+            graphics.Priority.Should().Be("base");
+            graphics.Depends.Should().BeNull();
+            graphics.Imports.Should().Be("grDevices");
+            graphics.LinkingTo.Should().BeNull();
+            graphics.Suggests.Should().BeNull();
+            graphics.Title.Should().Be("The R Graphics Package");
+            graphics.Author.Should().Be("R Core Team and contributors worldwide");
+        }
+
+        [Test]
+        [Category.PackageManager]
+        public async Task InstalledPackagesCustomLibPathNoPackages() {
+            // Setup library path to point to the test folder, don't install anything in it
+            using (var eval = await _workflow.RSession.BeginEvaluationAsync()) {
+                await SetLocalLibAsync(eval, _libPath);
+            }
+
+            var result = await _workflow.Packages.GetInstalledPackagesAsync();
+
+            // Since we can't remove Program Files folder from library paths,
+            // we'll get some results, but nothing from the test folder.
+            result.Should().NotBeEmpty();
+            result.Where(pkg => pkg.LibPath == _libPath.ToRPath()).Should().BeEmpty();
+        }
+
+        [Test]
+        [Category.PackageManager]
+        public async Task InstalledPackagesCustomLibPathOnePackage() {
+            // Setup library path to point to the test folder, install a package into it
+            using (var eval = await _workflow.RSession.BeginEvaluationAsync()) {
+                await SetLocalRepoAsync(eval, _repo1Path);
+                await SetLocalLibAsync(eval, _libPath);
+                await InstallPackageAsync(eval, TestPackages.RtvsLib1.Package, _libPath);
+            }
+
+            var result = await _workflow.Packages.GetInstalledPackagesAsync();
+
+            result.Should().NotBeEmpty();
+
+            var rtvslib1Pkg = result.SingleOrDefault(pkg => pkg.Package == TestPackages.RtvsLib1.Package && pkg.LibPath == _libPath.ToRPath());
+            rtvslib1Pkg.Should().NotBeNull();
+            rtvslib1Pkg.Version.Should().Be(TestPackages.RtvsLib1.Version);
+            rtvslib1Pkg.Depends.Should().Be(TestPackages.RtvsLib1.Depends);
+            rtvslib1Pkg.License.Should().Be(TestPackages.RtvsLib1.License);
+            rtvslib1Pkg.Built.Should().Be(TestPackages.RtvsLib1.Built);
+            rtvslib1Pkg.NeedsCompilation.Should().Be(null);
+            rtvslib1Pkg.Author.Should().Be(TestPackages.RtvsLib1.Author);
+            rtvslib1Pkg.Enhances.Should().Be(null);
+            rtvslib1Pkg.Title.Should().Be(TestPackages.RtvsLib1.Title);
+        }
+
+        private async Task SetLocalRepoAsync(IRSessionEvaluation eval, string localRepoPath) {
```

Comment:
I don't think the current implementation in the test will be appropriate for the product, let me have a chat with you on what you need.